src/provider/hyrax_pc.rs

```diff
@@ -0,0 +1,414 @@
+//! This module implements the Hyrax polynomial commitment scheme
+#![allow(clippy::too_many_arguments)]
+use crate::{
+  errors::SpartanError,
+  provider::ipa_pc::{InnerProductArgument, InnerProductInstance, InnerProductWitness},
+  provider::pedersen::{
+    Commitment as PedersenCommitment, CommitmentEngine as PedersenCommitmentEngine,
+    CommitmentEngineExtTrait, CommitmentKey as PedersenCommitmentKey,
+    CompressedCommitment as PedersenCompressedCommitment,
+  },
+  spartan::{
+    math::Math,
+    polynomial::{EqPolynomial, MultilinearPolynomial},
+  },
+  traits::{
+    commitment::{CommitmentEngineTrait, CommitmentTrait},
+    evaluation::EvaluationEngineTrait,
+    Group, TranscriptEngineTrait, TranscriptReprTrait,
+  },
+  Commitment, CommitmentKey,
+};
+use core::ops::{Add, AddAssign, Mul, MulAssign};
+use itertools::{
+  EitherOrBoth::{Both, Left, Right},
+  Itertools,
+};
+use rayon::prelude::*;
+use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
+
+/// A type that holds commitment generators for Hyrax commitments
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct HyraxCommitmentKey<G: Group> {
+  ck: PedersenCommitmentKey<G>,
+  _p: PhantomData<G>,
```

Comment:
The phantom type should not be needed since G constrains the CK.

Reply:
Fixed.